src/software/ai/navigator/navigator.cpp

```diff
@@ -6,6 +6,14 @@
 #include "software/ai/navigator/util.h"
 #include "software/ai/primitive/all_primitives.h"
 
+template <class... Ts>
```

Comment:
Maybe add a quick comment linking this to an example of `std::visit`? It's otherwise not super obvious what it's for and even then is pretty confusing code.

https://en.cppreference.com/w/cpp/utility/variant/visit

Reply:
done